crates/bevy_feathers/src/font_styles.rs

```diff
@@ -46,20 +47,27 @@ impl InheritableFont {
 /// propagates downward the font to all participating text entities.
 pub(crate) fn on_changed_font(
     insert: On<Insert, InheritableFont>,
-    font_style: Query<&InheritableFont>,
+    q_font_style: Query<&InheritableFont>,
+    q_children: Query<&Children>,
+    q_themed_text: Query<(), With<ThemedText>>,
     assets: Res<AssetServer>,
     mut commands: Commands,
 ) {
-    if let Ok(style) = font_style.get(insert.entity)
+    if let Ok(style) = q_font_style.get(insert.entity)
         && let Some(font) = match style.font {
             HandleOrPath::Handle(ref h) => Some(h.clone()),
             HandleOrPath::Path(ref p) => Some(assets.load::<Font>(p)),
         }
     {
-        commands.entity(insert.entity).insert(Propagate(TextFont {
-            font,
-            font_size: style.font_size,
-            ..Default::default()
-        }));
+        q_children
+            .iter_descendants(insert.entity)
+            .filter(|text_entity| q_themed_text.contains(*text_entity))
+            .for_each(|text_entity| {
+                commands.entity(text_entity).insert(TextFont {
+                    font: font.clone(),
+                    font_size: style.font_size,
+                    ..Default::default()
+                });
+            });
```

Comment:
Wouldn't this be better formed as a batch insert?

Reply:
Didn't know that was a thing.